### PR TITLE
SDK client changes for Traffic Manager API 2018-04-01

### PR DESCRIPTION
**Pull request by allencal**
## Description
Increment version to 2.5.0
New generated files and sessions.
New tests for custom subnets (IP address ranges), custom headers, expected status code ranges and multivalue.
Here is the REST spec PR: Azure/azure-rest-api-specs#3361
---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Review by dsgouda — CHANGES_REQUESTED** [review]
Please regenerate the code using generate.ps1 [here](https://github.com/allencal/azure-sdk-for-net/blob/bea232cc497bc5a36329df0b2da2c4514bca481b/src/SDKs/TrafficManager/Management.TrafficManager/generate.ps1)

**Comment by dsgouda** [comment]
@allencal please update the branch and commit the changes

**Comment by allencal** [comment]
All code in the 'generated' folder was updated with src\SDKs\TrafficManager\Management.TrafficManager\generate.ps1 (same as link to your comment), and committed.

**Comment by dsgouda** [comment]
@allencal the file diffs are unexpected. Did you have to copy over the files from any directory? Also, please update the branch

File: src/SDKs/TrafficManager/Management.TrafficManager/Microsoft.Azure.Management.TrafficManager.csproj [hunk 1]
```diff
@@ -1,25 +1,25 @@
...
+﻿<Project Sdk="Microsoft.NET.Sdk">
+	<!-- Please do not move/edit code below this line -->
+	<Import Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.reference.props'))" />
+	<!-- Please do not move/edit code below this line -->
+
+	<PropertyGroup>
+		<PackageId>Microsoft.Azure.Management.TrafficManager</PackageId>
+		<Description>Microsoft Azure Management TrafficManager Library</Description>
+		<AssemblyName>Microsoft.Azure.Management.TrafficManager</AssemblyName>
+		<Version>2.5.0</Version>
```
Comment (dsgouda) [inline src/SDKs/TrafficManager/Management.TrafficManager/Microsoft.Azure.Management.TrafficManager.csproj]: @allencal Please confirm whether you are ready to release a stable version

File: src/SDKs/TrafficManager/Management.TrafficManager/Microsoft.Azure.Management.TrafficManager.csproj [hunk 2]
```diff
@@ -1,25 +1,25 @@
...
+		<PackageTags>Microsoft Azure TrafficManager management;TrafficManager;TrafficManager management;</PackageTags>
+        <PackageReleaseNotes>
+            <![CDATA[
```
Comment (dsgouda) [inline src/SDKs/TrafficManager/Management.TrafficManager/Microsoft.Azure.Management.TrafficManager.csproj]: Please update `PackageReleaseNotes` here

File: src/SDKs/TrafficManager/Management.TrafficManager/Properties/AssemblyInfo.cs [hunk 3]
```diff
@@ -1,20 +1,20 @@
...
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Reflection;
+using System.Resources;
+using System.Runtime.InteropServices;
+
+[assembly: AssemblyTitle("Microsoft Azure Traffic Manager Management Library")]
+[assembly: AssemblyDescription("Provides Microsoft Azure Traffic Manager management functions for managing the Microsoft Azure Traffic Manager service.")]
+
+[assembly: AssemblyVersion("2.5.0.0")]
```
Comment (dsgouda) [inline src/SDKs/TrafficManager/Management.TrafficManager/Properties/AssemblyInfo.cs]: `AssemblyVersion` must be unchanged, `AssemblyFileVersion` must be updated

**Review by dsgouda — CHANGES_REQUESTED** [review]
Please run `msbuild build.proj /t:build /p:Scope=SDKs\TrafficManager` and commit changes to the .props file
Also, please merge all commits into a single commit and update the PR

File: src/SDKs/TrafficManager/Management.TrafficManager/Microsoft.Azure.Management.TrafficManager.csproj (hunk 1 above)
Comment (allencal) [reply src/SDKs/TrafficManager/Management.TrafficManager/Microsoft.Azure.Management.TrafficManager.csproj]: I am ready to release a stable version.

**Comment by dsgouda** [comment]
@allencal Looks good, please merge commits into a single commit and update the PR

**Comment by allencal** [comment]
Replacing with https://github.com/Azure/azure-sdk-for-net/pull/4576

**Pull request closed without merging**